heft-plugins/heft-dev-cert-plugin/src/ConfigureWebpackDevServerPlugin.ts

```diff
@@ -0,0 +1,128 @@
+            const certificateManager: CertificateManager = new CertificateManager();
+            await this._configureDevServerAsync(
+              webpackConfiguration,
+              certificateManager,
+              taskSession.logger,
+              devServerVersion
+            );
+            return webpackConfiguration;
+          }
+        );
+      }
+    );
+
+    taskSession.requestAccessToPluginByName(
+      '@rushstack/heft-webpack5-plugin',
+      Webpack5PluginName,
+      (accessor: IWebpack5PluginAccessor) => {
+        let devServerVersion: number | undefined;
+
+        accessor.onEmitWebpackVersionsHook!.tapPromise(
+          PLUGIN_NAME,
+          async (webpackVersions: IWebpack5Versions) => {
+            devServerVersion = this._determineMajorVersion(webpackVersions.webpackDevServerVersion);
+          }
+        );
+
+        accessor.onConfigureWebpackHook!.tapPromise(
+          PLUGIN_NAME,
+          async (webpackConfiguration: IWebpack5Configuration | null) => {
+            const certificateManager: CertificateManager = new CertificateManager();
+            await this._configureDevServerAsync(
+              webpackConfiguration,
+              certificateManager,
+              taskSession.logger,
+              devServerVersion
+            );
+            return webpackConfiguration;
+          }
+        );
+      }
+    );
+  }
+
+  private async _configureDevServerAsync(
+    webpackConfiguration: IWebpack4Configuration | IWebpack5Configuration | null,
+    certificateManager: CertificateManager,
+    logger: IScopedLogger,
+    webpackDevServerMajorVersion?: number
+  ): Promise<void> {
+    const certificate: ICertificate = await certificateManager.ensureCertificateAsync(true, logger.terminal);
+    if (!webpackConfiguration) {
+      logger.terminal.writeVerboseLine('No webpack configuration available to configure devServer.');
+    } else {
+      const configurations: (IWebpack4ConfigurationWithDevServer | IWebpack5ConfigurationWithDevServer)[] =
+        Array.isArray(webpackConfiguration) ? webpackConfiguration : [webpackConfiguration];
+      for (const configuration of configurations) {
+        if (webpackDevServerMajorVersion && webpackDevServerMajorVersion === 4) {
+          configuration.devServer = {
+            ...configuration.devServer,
+            server: {
+              type: 'https',
+              options: {
+                key: certificate.pemKey,
+                cert: certificate.pemCertificate
+              }
+            }
+          };
+        } else {
+          configuration.devServer = {
+            ...configuration.devServer,
+            https: {
+              key: certificate.pemKey,
+              cert: certificate.pemCertificate
+            }
+          };
+        }
+      }
+    }
+  }
+
+  private _determineMajorVersion(version?: string): number | undefined {
+    if (version) {
+      return parseInt(version);
```

Comment:
TIL `parseInt('4.5.1-1928019209dsoiajsdflka foobar') === 4`

Reply:
Yeah I had to double-check that when I was porting the code too. Just seems.. odd lol. But I see why it's done.